docs/DESKTOP_ENVIRONMENTS.md

```diff
@@ -0,0 +1,144 @@
+   instead of the caller starting the process and letting systemd know about it,
+   is encouraged.
+ * If no application ID is available, the launcher should generate a reasonable
+   name when possible (e.g. using `basename(argv[0])`). This name must not
+   contain a `-` character.
+
+This has the following advantages:
+ * Using the `apps-<launcher>-` prefix means that the unit defaults can be
+   adjusted using desktop environment specific drop-in files.
+ * The application ID can be retrieved by stripping the prefix and postfix.
+   This in turn should map to the corresponding `.desktop` file when available
+
+TODO: Define the name of slices that should be used.
+This could be `apps-<launcher>-<ApplicationID>-<RANDOM>.slice`, see also the
+cgroups section.
+
+TODO: Does it really make sense to insert the `<launcher>`? In GNOME I am
+currently using a drop-in to configure `BindTo=graphical-session.target`,
+`CollectMode=inactive-or-faield` and `TimeoutSec=5s`. I feel that such a
+policy makes sense, but it may make much more sense to just define a
+global default for all (graphical) applications.
+
+ * Should application lifetime be bound to the session?
+ * May the user have applications that do not belong to the graphical session (e.g. launched from SSH)?
+ * Could we maybe add a `apps-.service.d` dro
+
+## XDG autostart integration
+
+To allow XDG autostart integration, systemd will ship a cross-desktop generator
+to create appropriate units for the autostart directory.
+Desktop Environments will be able to make use of this simply by starting the
+appropriate XDG related targets (representing e.g. content of the
+`$XDG_CURRENT_DESKTOP` environment variable to handle `OnlyShowIn/NotShowIn`).
+The names and ordering rules for these targets are to be defined.
+
+This generator will likely never support certain desktop specific extensions.
+One such examle is the GNOME specific feature to bind a service to a settings
+variable.
+
+## Application managed cgroups
+
+Applications like browsers may place major components that are run as
+separate processes into into separate service or scope units.
+This can be useful to the applications themselves to apply further resource
+restrictions (e.g. disallow network access or memory limits).
+Furthermore, it exposes the structure of the application to the system,
+allowing it to e.g. kill entire cgroups in response to OOM situations
+without affecting the application as a whole.
+
+It is to be defined how these applications should create and manage these
+cgroups. Possible approaches could be:
+
+ * Directly use systemd and explicitly place items into a common slice.
+   This has the disadvantage of requiring privileged access to systemd.
+ * Provide an XDG portal, that allows creating relevant transient units.
+   This portal can enforce placement into the appropriate slices.
+ * Responsibility for management of cgroups may be delegated to applications,
+   see https://systemd.io/CGROUP_DELEGATION/.
```

Comment:
Once again, the part about privileged access make no sense. systemd --user does not require privileges, and we certainly don't want user applications to talk to the system manager.
Also, "common slice" is unclear. Maybe remove this whole block?

Reply:
Alright, I am fine with removing this in general. Should I leave the TODO with the possible implications or should I also drop that?

Reply:
Note that all this happens in a cgroup subtree owned by single uid, i.e. any resource control is just advisory, since a task can migrate itself elsewhere (in the tree). I'd document that apps can manage their subtree however needed as long as it happens under unit with `Delegate=` set.